nucypher/utilities/metrics.py

```diff
@@ -26,40 +77,223 @@ def collect_prometheus_metrics(ursula):
 
-def initialize_prometheus_exporter(ursula, port: int) -> None:
+    event_collectors_config = (
+        {
+            "collector": BaseEventMetricsCollector,
+            "name": "activity_confirmed", "contract_agent": staking_agent, "event": "ActivityConfirmed",
+            "argument_filters": {"staker": ursula.checksum_address},
+            "metrics": {"value": Gauge(f'{metrics_prefix}_activity_confirmed_value',
+                                       'Activity confirmed with value of locked tokens'),
+                        "period": Gauge(f'{metrics_prefix}_activity_confirmed_period',
+                                        'Activity confirmed period')}
+        },
+        {
+            "collector": BaseEventMetricsCollector,
+            "name": "mined", "contract_agent": staking_agent, "event": "Mined",
+            "argument_filters": {"staker": ursula.checksum_address},
+            "metrics": {"value": Gauge(f'{metrics_prefix}_mined_value', 'Mined value'),
+                        "period": Gauge(f'{metrics_prefix}_mined_period', 'Mined period'),
+                        "block_number": Gauge(f'{metrics_prefix}_mined_block_number', 'Mined block number')}
+        },
+        {
+            "collector": BaseEventMetricsCollector,
+            "name": "slashed_penalty", "contract_agent": staking_agent, "event": "Slashed",
+            "argument_filters": {"staker": ursula.checksum_address},
+            "metrics": {"penalty": Gauge(f'{metrics_prefix}_last_slashed_penalty', 'Penalty for slashing'),
+                        "block_number": Gauge(f'{metrics_prefix}_last_slashed_penalty_block_number',
+                                              'Slashed penalty block number')}
+        },
+        {
+            "collector": ReStakeEventMetricsCollector,
+            "name": "restake_set", "contract_agent": staking_agent, "event": "ReStakeSet",
+            "argument_filters": {"staker": ursula.checksum_address},
+            "metrics": {"reStake": Gauge(f'{metrics_prefix}_restaking', 'Restake set')}
+        },
+        {
+            "collector": WindDownEventMetricsCollector,
+            "name": "wind_down_set", "contract_agent": staking_agent, "event": "WindDownSet",
+            "argument_filters": {"staker": ursula.checksum_address},
+            "metrics": {"windDown": Gauge(f'{metrics_prefix}_wind_down', 'is windDown')}
+        },
+        {
+            "collector": WorkerSetEventMetricsCollector,
+            "name": "worker_set", "contract_agent": staking_agent, "event": "WorkerSet",
+            "argument_filters": {"staker": ursula.checksum_address},
+            "metrics": {"startPeriod": Gauge(f'{metrics_prefix}_worker_set_start_period', 'New worker was set'),
+                        "block_number": Gauge(f'{metrics_prefix}_worker_set_block_number', 'WorkerSet block number')}
+        },
+        {
+            "collector": BaseEventMetricsCollector,
```

Comment:
This function is pretty long (nearly 100 lines!) and has distinct partitions.  I suggest encapsulating each agent-local metrics generation.  This will be more modular, testable, readable, and reusable.